components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultURLResolverService.java

```diff
@@ -166,16 +193,29 @@ private void appendTenantPathParamInLegacyMode(StringBuilder serverUrl) {
         }
     }
 
-    private void appendTenantQueryParamInLegacyMode(StringBuilder serverUrl) {
+    private void appendTenantQueryParamInLegacyMode(StringBuilder serverUrl, String tenantDomain) {
 
-        String tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
+        if (StringUtils.isBlank(tenantDomain)) {
+            tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
+        }
```

Comment:
reason to not use the resolving method.
Looks like some checks/steps missed here